icu4j/main/classes/core/src/com/ibm/icu/impl/number/UsagePrefsHandler.java

```diff
@@ -0,0 +1,106 @@
+        final String kSuffixPrefix = "precision-increment/";
+        if (!precisionSkeleton.startsWith(kSuffixPrefix)) {
+            throw new IllegalIcuArgumentException("precisionSkeleton is only precision-increment");
+        }
+
+        String skeleton = precisionSkeleton.substring(kSuffixPrefix.length());
+        String skeletons[] = skeleton.split("/");
+        BigDecimal num = new BigDecimal(skeletons[0]);
+        BigDecimal den =
+                skeletons.length == 2 ?
+                        new BigDecimal(skeletons[1]) :
+                        new BigDecimal("1");
+
+
+        return Precision.increment(num.divide(den, MathContext.DECIMAL128));
+    }
+
+    protected static void mixedMeasuresToMicros(List<Measure> measures, DecimalQuantity quantity, MicroProps micros) {
+        if (measures.size() > 1) {
+            // For debugging
+            assert (micros.outputUnit.getComplexity() == MeasureUnit.Complexity.MIXED);
+
+            // Check that we received measurements with the expected MeasureUnits:
+            List<MeasureUnit> singleUnits =
+                    micros.outputUnit.splitToSingleUnits();
+
+            assert measures.size() == singleUnits.size();
+
+            // Mixed units: except for the last value, we pass all values to the
+            // LongNameHandler via micros->mixedMeasures.
+            for (int i = 0, n = measures.size(); i < n; i++) {
+                if (i == n - 1) {
+                    // The last value (potentially the only value) gets passed on via quantity.
+                    quantity.setToBigDecimal(BigDecimal.valueOf(measures.get(i).getNumber().doubleValue()));
+                    break;
+                }
+
+                micros.mixedMeasures.add(measures.get(i));
+            }
+        }
+    }
+
+    /**
+     * Obtains the appropriate output value, MeasureUnit and
+     * rounding/precision behaviour from the UnitsRouter.
+     * <p>
+     * The output unit is passed on to the LongNameHandler via
+     * micros.outputUnit.
+     */
+    @Override
+    public MicroProps processQuantity(DecimalQuantity quantity) {
+        fParent.processQuantity(quantity);
+
+
+        quantity.roundToInfinity(); // Enables toDouble
+        final UnitsRouter.RouteResult routed = fUnitsRouter.route(quantity.toBigDecimal());
+
+        MicroProps result = (MicroProps) this.fParent;
+        final List<Measure> routedMeasures = routed.measures;
+        result.outputUnit = routed.outputUnit.build();
+        result.mixedMeasures = new ArrayList<>();
+
+        UsagePrefsHandler.mixedMeasuresToMicros(routedMeasures, quantity, result);
+
+        String precisionSkeleton = routed.precision;
+        if (result.rounder != null) { /* TODO: it was before : result.rounder.fPrecision.isBogus() , is this correct */
+            if (precisionSkeleton.length() > 0) {
+                result.rounder = parseSkeletonToPrecision(precisionSkeleton);
```

Comment:
You should remember the rounding mode from the user's input settings.  @hugovdm

Reply:
C++ code:

    if (micros.rounder.fPrecision.isBogus()) {
        if (precisionSkeleton.length() > 0) {
            micros.rounder.fPrecision = parseSkeletonToPrecision(precisionSkeleton, status);
        } else {
            // We use the same rounding mode as COMPACT notation: known to be a
            // human-friendly rounding mode: integers, but add a decimal digit
            // as needed to ensure we have at least 2 significant digits.
            micros.rounder.fPrecision = Precision::integer().withMinDigits(2);
        }
    }

I'm guessing Java code might be something like this:

    if (micros.rounder == null) {
        if (precisionSkeleton.length() > 0) {
            micros.rounder.fPrecision = parseSkeletonToPrecision(precisionSkeleton);
        } else {
            ....
        }
    }

Reply:
Precision configuration comes from macrosToMicroGenerator() - in number_formatimpl.cpp - the default constructor produces a "bogus" Precision:

    // Rounding strategy
    Precision precision;
    if (!macros.precision.isBogus()) {
        precision = macros.precision;
    } else if (isCompactNotation) {
        precision = Precision::integer().withMinDigits(2);
    } else if (isCurrency) {
        precision = Precision::currency(UCURR_USAGE_STANDARD);
    } else if (macros.usage.isSet()) {
        // Bogus Precision - it will get set in the UsagePrefsHandler instead
        precision = Precision();
    } else {
        precision = Precision::maxFraction(6);
    }
    UNumberFormatRoundingMode roundingMode;
    roundingMode = macros.roundingMode;
    fMicros.rounder = {precision, roundingMode, currency, status};
    if (U_FAILURE(status)) {
        return nullptr;
    }

As discussed in the meeting: if this is too hard to resolve immediately, we can for now always use the UsagePreference skeleton, instead of only when the user didn't specify a Precision.